app/plugin.go

```diff
@@ -756,12 +756,22 @@ func (a *App) getPluginsFromFolder() (map[string]*pluginSignaturePath, *model.Ap
 		return nil, model.NewAppError("getPluginsFromDir", "app.plugin.sync.list_filestore.app_error", nil, appErr.Error(), http.StatusInternalServerError)
 	}
 
-	return getPluginsFromFilePaths(fileStorePaths), nil
+	return a.getPluginsFromFilePaths(fileStorePaths), nil
 }
 
-func getPluginsFromFilePaths(fileStorePaths []string) map[string]*pluginSignaturePath {
+func (a *App) getPluginsFromFilePaths(fileStorePaths []string) map[string]*pluginSignaturePath {
 	pluginSignaturePathMap := make(map[string]*pluginSignaturePath)
+
+	fsPrefix := ""
+	if *a.Config().FileSettings.DriverName == model.IMAGE_DRIVER_S3 {
+		ptr := a.Config().FileSettings.AmazonS3PathPrefix
+		if ptr != nil && *ptr != "" {
+			fsPrefix = *ptr + "/"
+		}
+	}
+
 	for _, path := range fileStorePaths {
+		path = strings.TrimPrefix(path, fsPrefix)
```

Comment:
@mickmister , I think the fix is ok, though alternatively one could factor out the trimming and S3 detection into one trimming wrapper function that has the advantage of isolating/removing the S3 check from the main flow and the disadvantage of having to do the check repeatedly.

Reply:
@catalintomai Yes I chose to do it this way due to having to check the config twice for each plugin. Do you think it's worth pulling out the code to create a function like:

```go
func (a *App) normalizePluginPath(pluginPath string) string
```

Reply:
@mickmister, for now I'd say it is ok, to fix the issue and close the incident, though it might be worth opening a ticket to investigate better handling of the issue ( maybe a generic map array function on the paths?)